Source/MLXFast/MLXFast.swift

```diff
@@ -55,11 +56,12 @@ public func RoPE(
 /// ```
 public func scaledDotProductAttention(
     queries: MLXArray, keys: MLXArray, values: MLXArray, scale: Float, mask: MLXArray?,
-    stream: StreamOrDevice = .default
+    memoryEfficientThreshold: Int = 1_000_000, stream: StreamOrDevice = .default
 ) -> MLXArray {
     MLXArray(
         mlx_fast_scaled_dot_product_attention(
-            queries.ctx, keys.ctx, values.ctx, scale, mask?.ctx, stream.ctx))
+            queries.ctx, keys.ctx, values.ctx, scale, mask?.ctx,
+            Int32(memoryEfficientThreshold), stream.ctx))
```

Comment:
Note: this is the case where `memoryEfficientThreshold` is `optional<int>` on the C++ side but `int` in the C layer.  The default value (for null) is not 0 or 1 but 1e6.  Given the API we have this is the best I can do (examine the implementation and hoist the value out of the code to here as the default).

@awni and @andresy -- per the offline conversation

Reply:
Note: we now have the optionals in the mlx-c API and this comment is obsolete